domain/src/commonMain/kotlin/com/islamversity/domain/repo/SurahListRepo.kt

```diff
@@ -0,0 +1,36 @@
+package com.islamversity.domain.repo
+
+import com.islamversity.core.Mapper
+import com.islamversity.core.listMap
+import com.islamversity.db.datasource.AyaLocalDataSource
+import com.islamversity.db.datasource.SurahLocalDataSource
+import com.islamversity.db.model.Calligraphy
+import com.islamversity.db.model.JuzEntity
+import com.islamversity.db.model.Surah
+import com.islamversity.domain.model.JuzRepoModel
+import com.islamversity.domain.model.SurahRepoModel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+
+interface SurahListRepo {
+    fun getAllSurah(): Flow<List<SurahRepoModel>>
+    fun getAllJuz(): Flow<List<JuzRepoModel>>
+}
+
+class SurahListRepoImpl(
+    private val surahDataSource: SurahLocalDataSource,
+    private val ayaDataSource: AyaLocalDataSource,
+    private val surahMapper: Mapper<Surah, SurahRepoModel>,
+    private val juzMapper: Mapper<JuzEntity, JuzRepoModel>
+) : SurahListRepo {
+
+    override fun getAllSurah(): Flow<List<SurahRepoModel>> =
+        surahDataSource.observeAllSurahs(Calligraphy("ar"))//TODO remove hardcoded Calligraphy
```

Comment:
please create a usecase on top and pass the calligraphy to this function

Reply:
How?

Reply:
A usecase like so:
```
class GetSurahsUsecase(
val surahRepo
val settingsRepo
){

fun getSurahs() = suraRepo.getAllSurah(settingsRepo.currentSurahCalligraphy())

}
```